include/zephyr/drivers/sdhc.h

```diff
@@ -242,6 +251,9 @@ __subsystem struct sdhc_driver_api {
 	int (*card_busy)(const struct device *dev);
 	int (*get_host_props)(const struct device *dev,
 				struct sdhc_host_props *props);
+	int (*enable_interrupt)(const struct device *dev,
+				sdhc_interrupt_cb_t callback, void *user_data);
+	int (*disable_interrupt)(const struct device *dev);
 };
```

Comment:
Hi- I am a bit confused what you are suggesting here. SD host controllers have one interrupt line. However, individual functions on an SDIO device can have interrupts, and each function interrupt is OR'ed together to issue an interrupt to the SD host controller.

Are you saying we should support enabling and disabling interrupts per each function within the SDIO stack? One way we could support this is to have functions like the following in the SDIO stack:
```
int sdio_enable_interrupt(struct sdio_func *func, sdio_interrupt_cb_t callback, void *user_data);
int sdio_disable_interrupt(struct sdio_func *func);
```

These functions could then only call the provided `callback` if an interrupt occurred for the given SDIO function

Reply:
Sorry for confusing, i mean  that current enable_interrupt/dissable_interrupt function operates only with Card interrupt
There are another interrupts:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/46566584/47aea88f-ad11-476d-be2e-c54f1f1a39bb)

If we do not want to handle them, maybe we can just rename:
enable_interrupt    -> enable_card_interrupt    
dissable_interrupt  -> dissable_card_interrupt

Reply:
Ah I understand- you mean additional interrupts in the host controller itself. Card removal wasn't something I considered since Zephyr does not support hotplugging devices. I'll do the following:
- rework the signature of the SDHC API to enable interrupt callbacks when an SDIO interrupt occurs, as well as when card insertion occurs
- rework the SDIO interrupt API to allow a consumer to register for interrupts from only one function source

Reply:
> rework the SDIO interrupt API to allow a consumer to register for interrupts from only one function source

After some additional consideration, I don't think we need an API for this feature yet. It won't really add anything, since SDHC drivers themselves can only register one callback at at time. Therefore, the SDIO subsystem would need a way to store callbacks for each function, which defeats the point of keeping SDIO function declarations out of the card structure (since we would now need callback pointers for each function). I will update the PR with a new signature for the SDHC API to allow for other card interrupts (like insertion and removal) to issue callbacks